app/services/course_service.py

```diff
@@ -60,4 +64,13 @@ async def get_instructor_gitea_organization_name(self) -> str:
     
     async def get_master_repository_name(self) -> str:
         course = await self.get_course()
-        return f"{ course.name }-class-master-repo"
\ No newline at end of file
+        return f"{ course.name }-class-master-repo"
+    
+    async def update_course(self, name: str, start_at: datetime, end_at: datetime) -> CourseModel:
```

Comment:
start_at, end_at is nullable elsewhere. Also, update_course should be broken into update_course_name, update_course_start_at, update_course_end_at

Reply:
done